cmd/agent/subcommands/run/command.go

```diff
@@ -516,28 +525,12 @@ func startAgent(
 		}
 	}
 
-	var logsAgentChannel chan *message.Message
-
-	// start logs-agent.  This must happen after AutoConfig is set up (via common.LoadComponents)
-	if pkgconfig.Datadog.GetBool("logs_enabled") || pkgconfig.Datadog.GetBool("log_enabled") {
-		if pkgconfig.Datadog.GetBool("log_enabled") {
-			log.Warn(`"log_enabled" is deprecated, use "logs_enabled" instead`)
-		}
-		if logsAgent, err := logs.Start(common.AC); err != nil {
-			log.Error("Could not start logs-agent: ", err)
-		} else {
-			logsAgentChannel = logsAgent.GetPipelineProvider().NextPipelineChan()
-		}
-	} else {
-		log.Info("logs-agent disabled")
-	}
-
 	// Start OTLP intake
 	otlpEnabled := otlp.IsEnabled(pkgconfig.Datadog)
 	inventories.SetAgentMetadata(inventories.AgentOTLPEnabled, otlpEnabled)
-	if otlpEnabled {
+	if logsAgent, ok := logsAgent.Get(); otlpEnabled && ok {
```

Comment:
Please watch out here, this component must start even if the logs agent is off, with a nil channel.

Reply:
got it, thanks